modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\falcon_donation\EventSubscriber;
+
+use Drupal\mailsystem\MailsystemManager;
+use Drupal\state_machine\Event\WorkflowTransitionEvent;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class FalconDonationOrderCompletedSubscriber.
+ *
+ * @package Drupal\falcon_donation\EventSubscriber
+ */
+class FalconDonationOrderCompletedSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The mail system manager.
+   *
+   * @var \Drupal\mailsystem\MailsystemManager
+   */
+  protected $pluginManagerMail;
+
+  /**
+   * A logger instance.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\mailsystem\MailsystemManager $plugin_manager_mail
+   *   The plugin manager mail.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The logger instance.
+   */
+  public function __construct(MailsystemManager $plugin_manager_mail, LoggerInterface $logger) {
+    $this->pluginManagerMail = $plugin_manager_mail;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events = [
+      'commerce_order.place.post_transition' => 'orderCompleteHandler',
+    ];
+    return $events;
+  }
+
+  /**
+   * Sending thank you email when order payment completed.
+   *
+   * @param \Drupal\state_machine\Event\WorkflowTransitionEvent $event
+   *   Order completed event.
+   */
+  public function orderCompleteHandler(WorkflowTransitionEvent $event) {
+    try {
+      /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
+      $order = $event->getEntity();
+      if ($order->bundle() !== 'donation') {
+        return;
+      }
+
+      if ($order->get('field_appeal')->isEmpty()) {
+        return;
+      }
+
+      $appeal = $order->get('field_appeal')->entity;
+
+      $subject = $appeal->get('field_thankyou_email_subject')->value;
+      $body = $appeal->get('field_thankyou_email_body')->value;
+
+      // Check subject and body for thank you email.
+      if (empty($subject) || empty($body)) {
```

Comment:
I don't believe that empty subject should cancel the email sending.

Reply:
Discussed.